docs/source/using-discovery/slurm.md

```diff
@@ -268,24 +264,100 @@ srun --partition=gpu --nodes=1 --ntasks=1 --gres=gpu:1 --mem=1Gb --time=01:00:00
 
 (job-arrays)=
-## Job Arrays
-Job arrays are a series of similar jobs. They are especially useful when you want to run the same job multiple times with minor changes, such as different input files.
-
-In HPC environments, users often need to run large numbers of jobs that are very similar, such as simulations with varying input parameters or the processing of multiple data files. Managing and tracking individual jobs can be a cumbersome and time-consuming process. **Enter: Slurm job arrays.**
+## Slurm Job Arrays
+Job arrays are a convenient way to submit and manage large numbers of similar jobs quickly. They can process millions of tasks in milliseconds, provided they are within size limits. Job arrays are particularly useful when running similar jobs, such as performing the same analysis with different inputs or parameters.
 
-A job array is a collection of related jobs submitted to Slurm as a single entity. A unique index identifies each job in the array, which runs independently on a separate compute node. The index can specify different input files or parameters for each job, allowing for the efficient processing of many similar tasks.
+Using job arrays can save time and reduce the amount of manual work required. Instead of submitting each job individually, you can submit a single job array and let Slurm handle the scheduling of individual jobs. This approach is beneficial if you have limited time or resources, as it allows you to use the cluster's computing power more efficiently by running multiple jobs in parallel.
 
 There are several ways to define job arrays, such as specifying the range of indices or providing a list of indices in a file. Slurm also offers various features to manage and track job arrays, such as options to simultaneously suspend, resume, or cancel all jobs in the array.
 
 ### Syntax: Job Arrays
+The most basic configuration for a job array is as follows:
+:::{code}
+#!/bin/bash
+#SBATCH --partition=general
+#SBATCH --job-name=jarray-example
+#SBATCH --output=out/array_%A.out
+#SBATCH --error=err/array_%A.err
+#SBATCH --array=1-6
+:::
+This command runs the same script six times using Slurm job arrays. Each job array has two additional environment variable sets. `SLURM_ARRAY_JOB_ID` (`%A`) is set to the first job ID of the array, and `SLURM_ARRAY_TASK_ID` (`%a`) is set to the job array index value.
+
+::::{tip}
+Generally, we want to pass the former as an argument for our script. If you are using R, you can retrieve the former using `task_id <- Sys.getenv("SLURM_ARRAY_TASK_ID")`. If you are using job arrays with Python, you can obtain the task ID using the following:
+
+:::{code} python
+import sys
+taskId = sys.getenv('SLURM_ARRAY_TASK_ID')
+:::
+::::
+
+In the previously defined Slurm header, the error and output file are overwritten whenever a "task" (one of the executions of the script through the job array) finishes. The modification below ensures that each "task id" will have its own output and error files.
```

Comment:
I agree with this but I think this can just be bundled into the above statement to make the 2 examples one.

Reply:
Nice! I did this and added a Note to callout the reason for using both (as it'll overwrite otherwise)